From a6dd49ac0789ae172b2e00b04a665a3dfce09590 Mon Sep 17 00:00:00 2001
From: Adriaan de Groot <groot@kde.org>
Date: Mon, 30 Oct 2023 00:37:38 +0100
Subject: [PATCH] packagechooser: rename #define for AppStreamQt

Make explicit that it is set to a version (if it is defined at all).
---
 src/modules/packagechooser/CMakeLists.txt  | 2 +-
 src/modules/packagechooser/Config.cpp      | 6 +++---
 src/modules/packagechooser/Tests.cpp       | 2 +-
 src/modules/packagechooserq/CMakeLists.txt | 2 +-
 4 files changed, 6 insertions(+), 6 deletions(-)

diff --git a/src/modules/packagechooser/CMakeLists.txt b/src/modules/packagechooser/CMakeLists.txt
index f42029964d..1f517fd33d 100644
--- a/src/modules/packagechooser/CMakeLists.txt
+++ b/src/modules/packagechooser/CMakeLists.txt
@@ -35,7 +35,7 @@ if(BUILD_APPSTREAM)
         TYPE OPTIONAL
     )
     if(AppStreamQt_FOUND)
-        add_definitions(-DHAVE_APPSTREAM)
+        add_definitions(-DHAVE_APPSTREAM_VERSION=${AppStreamQt_VERSION_MAJOR})
         list(APPEND _extra_libraries AppStreamQt)
         list(APPEND _extra_src ItemAppStream.cpp)
     endif()
diff --git a/src/modules/packagechooser/Config.cpp b/src/modules/packagechooser/Config.cpp
index 1d5b74f2c1..931071a4e9 100644
--- a/src/modules/packagechooser/Config.cpp
+++ b/src/modules/packagechooser/Config.cpp
@@ -14,7 +14,7 @@
 #include "ItemAppData.h"
 #endif
 
-#ifdef HAVE_APPSTREAM
+#ifdef HAVE_APPSTREAM_VERSION
 #include "ItemAppStream.h"
 #include <AppStreamQt/pool.h>
 #include <memory>
@@ -257,7 +257,7 @@ fillModel( PackageListModel* model, const QVariantList& items )
         return;
     }
 
-#ifdef HAVE_APPSTREAM
+#ifdef HAVE_APPSTREAM_VERSION
     std::unique_ptr< AppStream::Pool > pool;
     bool poolOk = false;
 #endif
@@ -284,7 +284,7 @@ fillModel( PackageListModel* model, const QVariantList& items )
         }
         else if ( item_map.contains( "appstream" ) )
         {
-#ifdef HAVE_APPSTREAM
+#ifdef HAVE_APPSTREAM_VERSION
             if ( !pool )
             {
                 pool = std::make_unique< AppStream::Pool >();
diff --git a/src/modules/packagechooser/Tests.cpp b/src/modules/packagechooser/Tests.cpp
index 1365b14524..533c070316 100644
--- a/src/modules/packagechooser/Tests.cpp
+++ b/src/modules/packagechooser/Tests.cpp
@@ -12,7 +12,7 @@
 #ifdef HAVE_APPDATA
 #include "ItemAppData.h"
 #endif
-#ifdef HAVE_APPSTREAM
+#ifdef HAVE_APPSTREAM_VERSION
 #include "ItemAppStream.h"
 #endif
 #include "PackageModel.h"
diff --git a/src/modules/packagechooserq/CMakeLists.txt b/src/modules/packagechooserq/CMakeLists.txt
index 52a89f88ec..b0b1f738dd 100644
--- a/src/modules/packagechooserq/CMakeLists.txt
+++ b/src/modules/packagechooserq/CMakeLists.txt
@@ -46,7 +46,7 @@ if(BUILD_APPSTREAM)
         TYPE OPTIONAL
     )
     if(AppStreamQt_FOUND)
-        add_definitions(-DHAVE_APPSTREAM)
+        add_definitions(-DHAVE_APPSTREAM_VERSION=${AppStreamQt_VERSION_MAJOR})
         list(APPEND _extra_libraries AppStreamQt)
         list(APPEND _extra_src ${_packagechooser}/ItemAppStream.cpp)
     endif()
